pg_search/src/index/writer/segment_component.rs

```diff
@@ -144,7 +144,10 @@ struct ExactBuffer<const CAPACITY: usize, W: Write> {
 
 impl<const CAPACITY: usize, W: Write> Drop for ExactBuffer<CAPACITY, W> {
     fn drop(&mut self) {
-        self.flush().ok();
+        // self.flush() creates buffers -- don't do this if we are not in a transaction i.e. have aborted
```

Comment:
IMO: let's add an assert to `Buffer::new()` (or whatever actually acquires the lock) to ensure that we don't acquire a lock outside of a transaction.

Then this conditional (and any other conditionals which were uncovered) would be about avoiding hitting that assert.

Reply:
If we do let’s make it a debug assert so that we don’t incur pgrx ffi overhead in release. We make a lot of buffers.